containers/unit_tests/TestVector.hpp

```diff
@@ -298,6 +298,36 @@ TEST(TEST_CATEGORY, vector_insert) {
   Impl::test_vector_insert<int, TEST_EXECSPACE>(3057);
 }
 
+TEST(TEST_CATEGORY, vector_push_back_default_exec) {
+  using TestVectorDeviceType =
+      Kokkos::Device<Kokkos::DefaultExecutionSpace,
+                     typename Kokkos::DefaultExecutionSpace::memory_space>;
+  using UInt64  = unsigned long long int;
+  using Pair    = Kokkos::pair<int, UInt64>;
+  using PairVec = Kokkos::vector<Pair, TestVectorDeviceType>;
```

Comment:
The bug was not sensitive to the value type was it?  If not then prefer a scalar type.